### DISCUSSION TITLE
Removing Pocketbase admin panel from custom build

### DISCUSSION THREAD

**Discussion by vipas84** [pocketbase/pocketbase#5970] [Q&A] [answered]
How to exclude PocketBase Svelte-based admin panel from build when using Pocketbase as a framework?

**Comment by ganigeorgiev** [comment]
There is no such option at the moment (_you can add a middleware that will intercept the ui requests but the file will be still embedded_). What actually is the use case?

Note that the UI is just a static client-side application, aka. if you are trying to prevent admins/superusers access, then hiding the UI doesn't really help much with this because the admins/superusers APIs would be still accessible.

**Reply by vipas84** [reply]
I develop an app where i need only few system settings to be exposed to the admin of the system (emails, backups + some custom ones), but not working with collections, records and viewing sensitive data (I'm aware that they could have access to it directly in the database). So basiclly I have my own limited admin panel in the design of my app to be embeded into the app... So I do not need to embed a bigger defaultre svelte admin panel to be a backdoor to extended capabilities with lack of branding... I could probabbly rewrite the route, but still there will be unneeded files in the binary

**Reply by ganigeorgiev** [reply, marked as answer]
We can add a build tag to exclude the files but the problem is that some parts of the app still require at the moment access to the UI (the default oauth2 redirect handler, the installer, etc.).

I've added it in my todo to think a little more on it but for now remains out of the scope.

**Reply by vipas84** [reply]
I just understood, that it could be a shortcut to make admin panel extendable/adjustable: 
- you do not need to rewrite it in Vuejs or Alpinejs...
- you could just publish in separate repository svelte-based admin panel (after  upgrade to Svelte5)
- provide a possibility to remove defaulr admin panel from a build (as an option)
- users could adjust existing admin panel and embed it by themselves if they need.
It could be a  good move towards PocketBase as a framework
